ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py

```diff
@@ -1467,3 +1468,33 @@ def test_torch_narrow(
         start=start,
         length=length,
     )
+
+
+@handle_frontend_test(
+    fn_tree="torch.diagonal_scatter",
+    dtype_and_values=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes(),
+        min_num_dims=2,
+        num_arrays=2,
+        shared_dtype=True,
+    ),

```

Comment:
Hey @suprasauce, you will need to expand the test to check all the arguments, including `offset`, `dim1` and `dim2`. To do so, I think it would be best to create an `@st.composite` strategy that generates everything.